libp2p/src/test/java/io/libp2p/transport/quic/QuicKuboTestJava.java

```diff
@@ -0,0 +1,89 @@
+package io.libp2p.transport.quic;
+
+import io.libp2p.core.Host;
+import io.libp2p.core.PeerId;
+import io.libp2p.core.Stream;
+import io.libp2p.core.StreamPromise;
+import io.libp2p.core.crypto.*;
+import io.libp2p.core.dsl.*;
+import io.libp2p.core.multiformats.*;
+import io.libp2p.protocol.*;
+import java.io.*;
+import java.net.*;
+import java.util.concurrent.*;
+import kotlin.*;
+import org.junit.jupiter.api.*;
+
+public class QuicKuboTestJava {
+  @Test
+  //    @Disabled
+  void pingKubo() throws Exception {
+    if (System.getProperty("os.name").toLowerCase().startsWith("windows")) return;
+    PeerId peerId = PeerId.fromBase58(getKuboPeerId());
+
+    Host clientHost =
+        new HostBuilder()
+            //                .secureTransport(QuicTransport::Ed25519)
+            .keyType(KeyType.ED25519)
+            .secureTransport(QuicTransport::Ecdsa)
+            .build();
+
+    CompletableFuture<Void> clientStarted = clientHost.start();
+    clientStarted.get(5, TimeUnit.SECONDS);
+    System.out.println("Client started");
+
+    Assertions.assertEquals(0, clientHost.listenAddresses().size());
+
+    StreamPromise<PingController> ping =
+        clientHost
+            .getNetwork()
+            .connect(peerId, new Multiaddr("/ip4/127.0.0.1/udp/4001/quic-v1"))
+            .thenApply(it -> it.muxerSession().createStream(new Ping()))
+            .get(5, TimeUnit.SECONDS);
+
+    Stream pingStream = ping.getStream().get(5, TimeUnit.SECONDS);
+    System.out.println("Ping stream created");
+    PingController pingCtr = ping.getController().get(5, TimeUnit.SECONDS);
+    System.out.println("Ping controller created");
+
+    for (int i = 0; i < 1000; i++) {
+      long latency = pingCtr.ping().get(1, TimeUnit.SECONDS);
+      System.out.println("Ping is " + latency);
+    }
+    pingStream.close().get(5, TimeUnit.SECONDS);
+    System.out.println("Ping stream closed");
+
+    Assertions.assertThrows(
+        ExecutionException.class, () -> pingCtr.ping().get(5, TimeUnit.SECONDS));
+
+    clientHost.stop().get(5, TimeUnit.SECONDS);
+    System.out.println("Client stopped");
+  }
+
+  private static String getKuboPeerId() throws IOException, URISyntaxException {
+    String url = "http://localhost:5001/api/v0/id";
```

Comment:
why this is running on windows while the previous isnt?

Reply:
The install script only handle linux, no point adding windows. So the test is disabled on windows.

Reply:
ahh sorry this is not a test, it is just a function.
You can remove the `//    @Disabled` up there.